docker/Dockerfile

```diff
@@ -0,0 +1,159 @@
+# syntax=docker/dockerfile:1.2
+#################################################
+#
+# Create base image with python installed.
+#
+# DL3007 ignored because base-docker we specifically always want to build on
+# the latest base image, by design.
+#
+# hadolint ignore=DL3007
+FROM ghcr.io/opensafely-core/base-docker:latest as base-python
+
+# we are going to use an apt cache on the host, so disable the default debian
+# docker clean up that deletes that cache on every apt install
+RUN rm -f /etc/apt/apt.conf.d/docker-clean
+
+# ensure fully working base python3 installation
+# see: https://gist.github.com/tiran/2dec9e03c6f901814f6d1e8dad09528e
+# use space efficient utility from base image
+RUN --mount=type=cache,target=/var/cache/apt \
+  /root/docker-apt-install.sh python3 python3-venv python3-pip python3-distutils tzdata ca-certificates
+
+# install any additional system dependencies
+COPY docker/dependencies.txt /tmp/dependencies.txt
+RUN --mount=type=cache,target=/var/cache/apt \
+    /root/docker-apt-install.sh /tmp/dependencies.txt
```

Comment:
as an aside - I'm not really into the overloading of the argument(s) to `docker-apt-install.sh`, in which they can be filenames or package names - it's a little too "magical" for my taste - but that's for the other repo not for this one!

Reply:
right, it's a bash script and I didn't want to do proper argument parsing. I'm ok with it, but happy to switch if you prefer (although we'd need to update all the images that use it too)

Reply:
I don't think it's particularly worth changing, mostly mentioned it for kind of calibrating our judgements with each other for future choices